docs/contributing.md

```diff
@@ -2,6 +2,12 @@
 
 We welcome contributions from the community! This guide will help you understand our contribution process and requirements.
 
+## Development guidelines
+
+1. Small PRs ([blogpost](https://testing.googleblog.com/2024/07/in-praise-of-small-pull-requests.html))
+1. If you are fixing a bug, write a test reproducing a bug (the test should fail without your changes)
+1. If you are refactoring, make sure there are tests covering this area, if not - introduce tests first, then continue with refactoring
```

Comment:
```suggestion
1. If you are refactoring, ensure adequate test coverage exists for the target area. If coverage is insufficient, create tests in a separate pull request first. This approach provides a safety net for validating current behavior and simplifies code reviews.
```

Reply:
@artemrys regarding wording I like that one :D